File: JavaScript/111-Minimum Depth of Binary Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var minDepth = function(root) {
    if (!root) return 0;
    
    let queue = [root];
    let depth = 0
    while (queue.length){
        depth += 1
        let size = queue.length;
        for (let i=0; i<size; i++){
            let curr = queue.shift();
            if (!curr.left && !curr.right){
                return depth;
            }
            if (curr.left){
                queue.push(curr.left);
            }
            if (curr.right){
                queue.push(curr.right);
            }
        }
    }
};

File: 111-Minimum Depth of Binary Tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def minDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if not root:
            return 0
        
        queue = collections.deque([(root,1)])
        
        while queue:
            for _ in range(len(queue)):
                curr, level = queue.popleft()
                if not curr.left and not curr.right:
                    return level
                if curr.left:
                    queue.append((curr.left, level+1))
                if curr.right:
                    queue.append((curr.right, level+1))